### DISCUSSION TITLE
Connecting GitHub Accounts to NPM

### DISCUSSION THREAD

**Discussion by grempe** [npm/feedback#738] [Website] [answered]
In the new blog post [Introducing even more security enhancements to NPM](https://github.blog/2022-07-26-introducing-even-more-security-enhancements-to-npm/) you announced the ability to link GitHub, and Twitter, profile info to an NPM account profile for display as an authenticated profile.

This is great, but there is one issue I would like to see addressed.  If you have an NPM organization, it would be more desirable to link that org to a GitHub **organization**, and not an individual GitHub **account**.  I would want the NPM profile to show the info for, and link to, my GitHub organization and not the owner or a member of that org.

You only allow linking to a specific GitHub account now, which will mean that I probably won't be linking this NPM org since it is not a single user account profile that I want to prove or highlight. I would prefer for you to prove that a GitHub org owner level account is linked to NPM, but display the profile info for the GitHub org and not that user on NPM.

**Comment by MylesBorins** [comment, marked as answer]
Being able to link a GitHub Organization to an npm organization is a great idea, but not something we are currently planning to support.

The primary reason we've invested in the feature at this time was for the purpose of identity verification for account recovery, there is the obvious added benefit that public data is now verified but that was not the original goal.

Our next phase of related work is going to be in more features related to individual identity verification and streamlining the recovery process, which in itself serves our larger goal of enforcing 2FA to high-impact project. More 2FA means more account recovery, which has a significant human cost to developers and our support agents.

While I recognize that verifying organizations are who they say they are is an important problem to solve, it is not one we are currently prioritizing. If we do visit it in the future the solution may look quite a bit different than the account linking. That said I definitely appreciate the feedback!

**Reply by grempe** [reply]
If the primary reason for the feature is account recovery  (very reasonable) I would suggest that the blog post is not in alignment with that. The first paragraph describing the feature (and implicitly the primary reason) states:

> Linking of these accounts is performed via official integrations with both GitHub and Twitter and ensures that verified account data are included on npm profiles moving forward. We will no longer be showing the previously unverified GitHub or Twitter data on public user profiles, making it possible for developers to audit identities and trust that an account is who they say they are.

This was my takeaway from the post. Better, more auditable, NPM profile info. For organizations on NPM, it might actually be a step backward IMHO since they may be more reluctant to have any GitHub account linkage since they'd have to choose a user account who may be unknown to interested viewers, or change over time.

**Reply by MylesBorins** [reply]
@grempe you can blame me then, I authored it 😓 

The intention with the blog post was to surface to customers the net impact on user experience, but to your point it resulted in burying the lede, will aim to do better next time

**Reply by grempe** [reply]
All good. Its not a big deal. Just wanted to pass along some feedback.

There's always feedback. 😀

Cheers.
